lib/WorseReflection/Core/Inference/Resolver/MemberAccessExpressionResolver.php

```diff
@@ -23,16 +24,23 @@
 
 class MemberAccessExpressionResolver implements Resolver
 {
+    private GenericTypeResolver $genericResolver;
+
+    public function __construct()
+    {
+        $this->genericResolver = new GenericTypeResolver();
+    }
+
     public function resolve(NodeContextResolver $resolver, Frame $frame, Node $node): NodeContext
     {
         assert($node instanceof MemberAccessExpression);
 
         $class = $resolver->resolveNode($frame, $node->dereferencableExpression);
 
-        return self::infoFromMemberAccess($resolver, $frame, $class->type(), $node);
+        return self::infoFromMemberAccess($resolver, $frame, $class->type(), $node, $this->genericResolver);
     }
 
-    public static function infoFromMemberAccess(NodeContextResolver $resolver, Frame $frame, Type $classType, Node $node): NodeContext
+    public static function infoFromMemberAccess(NodeContextResolver $resolver, Frame $frame, Type $classType, Node $node, ?GenericTypeResolver $genericResolver = null): NodeContext
```

Comment:
this static method was a hack to share the functionality with the scoped member access resolver. Probably "fine" for now.